app/src/main/java/io/github/notefydadm/notefy/model/Note.java

```diff
@@ -62,6 +65,10 @@ public Note(String title, String userID, NoteState state, boolean isFavorite, @C
         this.blocks = blocks;
```

Comment:
```suggestion
        if (blocks == null) {
            this.blocks = new ArrayList<>();
        } else {
            this.blocks = blocks;
        }
```
Lo que decía en el comentario de arriba. Comprobar si es `null`, para que cuando en tu clase llames al constructor con blocks valiendo `null`, no pete luego.